File: csharp/VeriPaketi.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace NDP_Odev4
{
    internal class VeriPaketi : KuantumNesnesi
    {
        public VeriPaketi(string id, int stabilite, int tseviyesi):base(id, stabilite, tseviyesi)
        {
            NesneID = id;
            Stabilite = stabilite;
            TehlikeSeviyesi = tseviyesi;
        }

        public override void AnalizEt()
        {
            int yeniDeger = Stabilite - 5;
            

            if (yeniDeger <= 0)
            {
                throw new KuantumCokusuException(NesneID);
            }
            Stabilite = yeniDeger;
            Console.WriteLine("Veri icerigi okundu...");

        }
    }
}

File: javascript/VeriPaketi.js
const KuantumNesnesi = require('./KuantumNesnesi');
const KuantumCokusuException = require('./KuantumCokusuException');

class VeriPaketi extends KuantumNesnesi {
    constructor(id, stabilite, tseviyesi) {
        super(id, stabilite, tseviyesi);
    }

    AnalizEt() {
        let yeniDeger = this.Stabilite - 5;
        if (yeniDeger <= 0) {
            throw new KuantumCokusuException(this.NesneID);
        }
        this.Stabilite = yeniDeger;
        console.log("Veri icerigi okundu...");
    }
}

module.exports = VeriPaketi;